### PR TITLE
feat: messaging controller commands

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- created controller command packages which contains business logic and
models needed for execution of all controller API commands, including
REST endpoints and upcoming JS Worker commands (#1070).

- added command errors, which can be used by REST and JS Worker(WASM)
for handling/sending errors.

- Part of #1175

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: pkg/controller/command/errors.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// Note: recommended to use [0-9]*000 pattern for any new entries
+// Example: 2000, 3000, 4000 ...... 25000
+type Group int32
+
+const (
+	// Common error group for general command errors
+	Common Group = 1000
+
+	// DIDExchange error group for DID exchange protocol command errors
+	DIDExchange Group = 2000
```
Comment (troyronda) [inline pkg/controller/command/errors.go]: We shouldn't be referencing specific protocols.

Comment (sudeshrshetty) [reply pkg/controller/command/errors.go]: these are specific to command types, I will fix the comment.

Comment (sudeshrshetty) [reply pkg/controller/command/errors.go]: fixed

Comment (troyronda) [reply pkg/controller/command/errors.go]: My point is actually that DIDExchange nor any other specific protocol/message should not appear in this file.

Comment (troyronda) [reply pkg/controller/command/errors.go]: I guess this is existing code, but I don't really understand why we didn't use a namespace string defined by individual command packages.

**Pull request merged**